rust/feature-flags/src/flag_definitions.rs

```diff
@@ -7,11 +7,6 @@ use tracing::instrument;
 // TODO: Add integration tests across repos to ensure this doesn't happen.
 pub const TEAM_FLAGS_CACHE_PREFIX: &str = "posthog:1:team_feature_flags_";
 
-// TODO: Hmm, revisit when dealing with groups, but seems like
-// ideal to just treat it as a u8 and do our own validation on top
-#[derive(Debug, Deserialize)]
-pub enum GroupTypeIndex {}
-
```

Comment:
We don't need to do our own validation on top anymore, found out you can have an enum with u8 which is exactly what we want here https://serde.rs/enum-number.html

Reply:
I bring this up later, but given that Postgres uses an INT type instead of an [ENUM](https://www.postgresql.org/docs/current/datatype-enum.html), I'm going to just have this field be a `i32`, and I'll include to comment to explain that it's a bounded enum.  Alternatively, I could write a migration for the postgres table to change this field, and then model it in Rust accordingly.  Seems the most correct but likely overkill.  My opinion is to go with i32, add a comment explaining the constraints, and then do a migration down the road.

Reply:
sounds good

Reply:
You might want to see [how I handle group-type-indexes in propdefs](https://github.com/PostHog/posthog/blob/master/rust/property-defs-rs/src/types.rs#L70). Resolving them is pretty cheap, [and I use a small concurrent cache (stored in the app_context) to avoid doing it where I can](https://github.com/PostHog/posthog/blob/master/rust/property-defs-rs/src/app_context.rs#L84)